internal/verification/database/health_authority_db.go

```diff
@@ -0,0 +1,157 @@
+	if len(ha.Keys) != 0 {
+		return fmt.Errorf("unable to insert health authority with keys, attach keys later")
+	}
+	if err := ha.Validate(); err != nil {
+		return err
+	}
+
+	return db.db.InTx(ctx, pgx.Serializable, func(tx pgx.Tx) error {
+		row := tx.QueryRow(ctx, `
+			INSERT INTO
+				HealthAuthority
+				(iss, aud, name)
+			VALUES
+				($1, $2, $3)
+			RETURNING id
+			`, ha.Issuer, ha.Audience, ha.Name)
+		if err := row.Scan(&ha.ID); err != nil {
+			return fmt.Errorf("inserting healthauthority: %w", err)
+		}
+		return nil
+	})
+}
+
+// GetHealthAuthority retrieves a HealthAuthority record by the issuer name.
+func (db *HealthAuthorityDB) GetHealthAuthority(ctx context.Context, issuer string) (*model.HealthAuthority, error) {
+	conn, err := db.db.Pool.Acquire(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("acquiring connection: %w", err)
+	}
+	defer conn.Release()
+
+	row := conn.QueryRow(ctx, `
+		SELECT
+			id, iss, aud, name
+		FROM
+			HealthAuthority
+		WHERE
+			iss = $1`, issuer)
+
+	var ha model.HealthAuthority
+	if err := row.Scan(&ha.ID, &ha.Issuer, &ha.Audience, &ha.Name); err != nil {
+		return nil, err
+	}
+
+	haks, err := db.GetHealthAuthorityKeys(ctx, &ha)
+	if err != nil {
+		return nil, err
+	}
+	ha.Keys = haks
+
+	return &ha, nil
+}
+
+func (db *HealthAuthorityDB) AddHealthAuthorityKey(ctx context.Context, ha *model.HealthAuthority, hak *model.HealthAuthorityKey) error {
+	if err := hak.Validate(); err != nil {
+		return err
+	}
+	if ha.ID == 0 {
+		return errors.New("invalid health authority ID, must be non zero")
+	}
+
+	hak.AuthorityID = ha.ID
+	thru := db.db.NullableTime(hak.Thru)
+	return db.db.InTx(ctx, pgx.Serializable, func(tx pgx.Tx) error {
+		result, err := tx.Exec(ctx, `
+			INSERT INTO
+				HealthAuthorityKey
+				(health_authority_id, version, from_timestamp, thru_timestamp, public_key)
+			VALUES
+				($1, $2, $3, $4, $5)
+			`, hak.AuthorityID, hak.Version, hak.From, thru, hak.PublicKeyPEM)
+		if err != nil {
+			return fmt.Errorf("inserting healthauthoritykey: %w", err)
+		}
+		if result.RowsAffected() != 1 {
+			return fmt.Errorf("no rows inserted")
+		}
+		return nil
+	})
+}
+
+func (db *HealthAuthorityDB) GetHealthAuthorityKeys(ctx context.Context, ha *model.HealthAuthority) ([]*model.HealthAuthorityKey, error) {
+	conn, err := db.db.Pool.Acquire(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("acquiring connection: %w", err)
+	}
+	defer conn.Release()
+
+	rows, err := conn.Query(ctx, `
+		SELECT
+		  health_authority_id, version, from_timestamp, thru_timestamp, public_key
+		FROM
+		  HealthAuthorityKey
+		WHERE
+		  health_authority_id = $1`, ha.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var haks []*model.HealthAuthorityKey
+	for rows.Next() {
+		if rows.Err() != nil {
+			return nil, rows.Err()
+		}
+		var hak model.HealthAuthorityKey
+		var thru *time.Time
+		if err := rows.Scan(&hak.AuthorityID, &hak.Version, &hak.From, &thru, &hak.PublicKeyPEM); err != nil {
```

Comment:
Maybe extract this into a `scanOneHAKey` heklper like we did elsewhere?

Reply:
will do if I add another query method. Right now this is the only one.